src/utils/hookRunner.js

```diff
@@ -15,7 +21,8 @@ class hookRunner {
       || this.hooks[taskName].length === 0
     ) { return null; }
 
-    // Manual looping is done to prevent passing function in hooks[taskName] into a callback
+    // Manual looping is done to prevent passing function in hooks[taskName] into a callback,
+    // which might cause scoping issues
     for (let idx = 0; idx < this.hooks[taskName].length; idx += 1) {
       if (this.hooks[taskName][idx].constructor.name === 'AsyncFunction') {
```

Comment:
No need for this, you can `await` on a non-async function.

Reply:
removed